### PR TITLE
Document the support policy for optional packages

### PR DESCRIPTION
**Pull request by seisman**
**Preview**: https://pygmt-dev--3616.org.readthedocs.build/en/3616/minversions.html

Closes https://github.com/GenericMappingTools/pygmt/issues/3456.

### CODE REVIEW

File: doc/minversions.md [hunk 1]
```diff
@@ -20,12 +20,16 @@ myst:
...
-- Support for Python versions be dropped 3 years after their initial release.
-- Support for core package dependencies (NumPy, pandas, Xarray) be dropped 2 years after
-  their initial release.
+- Drop support for Python versions 3 years after their initial release.
+- Drop support for core package dependencies (NumPy, pandas, Xarray) 2 years after their
+  initial release.
 
-Similarly, the PyGMT team has decided to discontinue support for GMT versions 3 years
-after their initial release.
+In accordance with this policy, the PyGMT team has also decided:
```
Comment (seisman) [inline doc/minversions.md:27]:
```suggestion
In accordance with this policy, the PyGMT team has also decided to:
```

File: doc/minversions.md [hunk 2]
```diff
@@ -20,12 +20,16 @@ myst:
 PyGMT has adopted [SPEC 0](https://scientific-python.org/specs/spec-0000/) alongside the
 rest of the scientific Python ecosystem, and therefore:
```
Comment (weiji14) [inline doc/minversions.md:21]:
```suggestion
rest of the scientific Python ecosystem, and will therefore:
```

File: doc/minversions.md [hunk 3]
```diff
@@ -20,12 +20,16 @@ myst:
...
-- Support for Python versions be dropped 3 years after their initial release.
-- Support for core package dependencies (NumPy, pandas, Xarray) be dropped 2 years after
-  their initial release.
+- Drop support for Python versions 3 years after their initial release.
+- Drop support for core package dependencies (NumPy, pandas, Xarray) 2 years after their
+  initial release.
 
-Similarly, the PyGMT team has decided to discontinue support for GMT versions 3 years
-after their initial release.
+In accordance with this policy, the PyGMT team has also decided to:
```
Comment (weiji14) [inline doc/minversions.md:27]: Maybe word it like this? It sounds strange to say "in accordance with this [SPEC 0] policy"
```suggestion
In addition to the above, the PyGMT team has also decided to:
```

File: doc/minversions.md [hunk 4]
```diff
@@ -20,12 +20,16 @@ myst:
...
+
+- Drop support for GMT versions 3 years after their initial release, while ensuring at
+  least two latest minor versions remain supported.
```
Comment (seisman) [inline doc/minversions.md:30]: > ensuring at least two latest minor versions remain supported.

I've added this sentence to the GMT support policy.

File: doc/minversions.md [hunk 5]
```diff
@@ -20,12 +20,16 @@ myst:
...
+- Maintain support for optional dependencies for 1 year only. Users are encouraged to
+  use the most up-to-date optional dependencies where possible.
```
Comment (weiji14) [inline doc/minversions.md:32]: Trying to get the wording right...

```suggestion
- Maintain support for optional dependencies for at least 1 year after their initial
  release. Users are encouraged to use the most up-to-date optional dependencies where
  possible.
```

**Review by weiji14 — COMMENTED** [review]
Also debating on whether to add something based on this note from SPEC 0:

> Core packages may or may not decide to provide bug fix releases during the full 2 year period after release. Therefore, projects may occasionally want to drop support for core package dependencies earlier than recommended by this SPEC. For instance, if a newer minimum version of a core package is needed by a project due to a critical bug fix, which is not backported to older versions.

E.g. we have pinned to xarray>=2023.04 at https://github.com/GenericMappingTools/pygmt/pull/3460 earlier than expected.

**Comment by seisman** [comment]
> Also debating on whether to add something based on this note from SPEC 0:
> 
> > Core packages may or may not decide to provide bug fix releases during the full 2 year period after release. Therefore, projects may occasionally want to drop support for core package dependencies earlier than recommended by this SPEC. For instance, if a newer minimum version of a core package is needed by a project due to a critical bug fix, which is not backported to older versions.
> 
> E.g. we have pinned to xarray>=2023.04 at #3460 earlier than expected.

Maybe add a note like:

> The policy is not strictly enforced and occasional breaches of the policy are allowed after discussions within the PyGMT team.

**Review by michaelgrund — APPROVED** [review]

File: doc/minversions.md [hunk 6]
```diff
@@ -18,14 +18,24 @@ myst:
...
 
-Similarly, the PyGMT team has decided to discontinue support for GMT versions 3 years
-after their initial release.
+In addition to the above, the PyGMT team has also decided to:
+
+- Drop support for GMT versions 3 years after their initial release, while ensuring at
+  least two latest minor versions remain supported.
+- Maintain support for optional dependencies for at least 1 year after their initial
+  release. Users are encouraged to use the most up-to-date optional dependencies where
+  possible.
```
Comment (weiji14) [inline doc/minversions.md:33]: Maybe link to https://pygmt-dev--3616.org.readthedocs.build/en/3616/ecosystem.html#pygmt-dependencies so people know what the optional dependencies are.

```suggestion
- Maintain support for [optional dependencies](/ecosystem.md#pygmt-dependencies) for at
  least 1 year after their initial release. Users are encouraged to use the most
  up-to-date optional dependencies where possible.
```

File: doc/minversions.md [hunk 7]
```diff
@@ -18,14 +18,24 @@ myst:
...
+
+:::{note}
+The policy is not strictly enforced and occasional breaches of the policy are allowed
+after discussions within the PyGMT team.
```
Comment (weiji14) [inline doc/minversions.md:37]: I think we should be specific that breaches to the policy are only for **core** and **optional** packages following the SPEC 0 note (i.e. Python and GMT versions are to be dropped strictly after 3 years). How about wording it like this?

```suggestion
The SPEC 0 policy is enforced on a best-effort basis, and the PyGMT team may decide to
drop support for core (and optional) package dependencies earlier than recommended for
compatibility reasons.
```

**Review by weiji14 — APPROVED** [review]

**Pull request merged**